### PR TITLE
Enabling cron expression on nightly builds

### PR DESCRIPTION
**Pull request by nikhilumeshsargur**
*Issue #, if available:*

*Description of changes:* Enabling cron expression on nightly builds

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by dylanraws — APPROVED** [review]

File: src/package_staleness.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def _generate_report(package_versions_in_upstream, target_packages_match_spec_ou
             print(package + '|' + version_in_sagemaker_distribution + '|' +
                   package_versions_in_upstream[package])
         else:
-            print('${\color{red}' + package + '}$' + '|' + version_in_sagemaker_distribution +
+            print('${color{red}' + package + '}$' + '|' + version_in_sagemaker_distribution +
```
Comment (claytonparnell) [inline src/package_staleness.py:66]: This might mess with the staleness report.. I know it throws a warning for python syntax, but it uses the Markdown/Latex syntax for github comment. We should validate that it would still look ok in a github comment

File: .github/workflows/nightly-build-image.yml [hunk 2]
```diff
@@ -1,10 +1,8 @@
...
-on: workflow_dispatch
-#on:
-##  # Run every night at 5:15PM PST
-##  # Run before monthly, so we don't immediately patch a new minor version
-##  schedule:
-##    - cron: '0 15 17 ? * *'
+on: 
+  # Run every night at 4:00PM PST
+  schedule:
+    - cron: '0 16 ? * *'
```
Comment (claytonparnell) [inline .github/workflows/nightly-build-image.yml:5]: I believe this is UTC time. So this cron would run in the morning PST

File: src/package_staleness.py (hunk 1 above)
Comment (nikhilumeshsargur) [reply src/package_staleness.py:66]: # Staleness Report: 1.6.1(gpu)

Package | Current Version in the Distribution image | Latest Relevant Version in Upstream
---|---|---
numpy|1.26.4|1.26.4
jinja2|3.1.3|3.1.3
pandas|2.1.4|2.1.4
altair|5.2.0|5.2.0
${color{red}boto3}$|1.34.51|1.34.69
ipython|8.22.2|8.22.2
jupyter-lsp|2.2.4|2.2.4
jupyterlab|4.1.5|4.1.5
amazon-codewhisperer-jupyterlab-ext|2.0.1|2.0.1
jupyter-scheduler|2.5.1|2.5.1
amazon-sagemaker-jupyter-scheduler|3.0.7|3.0.7
${color{red}amazon-sagemaker-sql-magic}$|0.1.0|0.1.1
jupyterlab-lsp|5.0.3|5.0.3
${color{red}amazon_sagemaker_sql_editor}$|0.1.3|0.1.4
scipy|1.11.4|1.11.4
scikit-learn|1.4.1.post1|1.4.1.post1
pip|23.3.2|23.3.2
torchvision|0.15.2|0.15.2
autogluon|0.8.2|0.8.2
ipywidgets|8.1.2|8.1.2
notebook|7.1.2|7.1.2
aws-glue-sessions|1.0.4|1.0.4
conda|23.11.0|23.11.0
fastapi|0.103.2|0.103.2
${color{red}langchain}$|0.1.9|0.1.13
jupyter-ai|2.11.0|2.11.0
jupyter-dash|0.4.2|0.4.2
jupyter-server-proxy|4.1.2|4.1.2
jupyterlab-git|0.50.0|0.50.0
keras|2.12.0|2.12.0
matplotlib|3.8.3|3.8.3
nodejs|18.19.0|18.19.0
py-xgboost-gpu|1.7.6|1.7.6
thrift_sasl|0.4.3|0.4.3
pyhive|0.7.0|0.7.0
python-gssapi|1.8.3|1.8.3
python-lsp-server|1.10.1|1.10.1
pytorch-gpu|2.0.0|2.0.0
sagemaker-code-editor|1.0.0|1.0.0
sagemaker-headless-execution-driver|0.0.12|0.0.12
sagemaker-jupyterlab-emr-extension|0.1.9|0.1.9
sagemaker-jupyterlab-extension|0.2.0|0.2.0
sagemaker-kernel-wrapper|0.0.2|0.0.2
sagemaker-python-sdk|2.212.0|2.212.0
sagemaker-studio-analytics-extension|0.0.21|0.0.21
sasl|0.3.1|0.3.1
supervisor|4.2.5|4.2.5
tensorflow|2.12.1|2.12.1
uvicorn|0.28.1|0.28.1

# Staleness Report: 1.6.1(cpu)

Package | Current Version in the Distribution image | Latest Relevant Version in Upstream
---|---|---
numpy|1.26.4|1.26.4
jinja2|3.1.3|3.1.3
pytorch|2.0.0|2.0.0
pandas|2.1.4|2.1.4
altair|5.2.0|5.2.0
${color{red}boto3}$|1.34.51|1.34.69
ipython|8.22.2|8.22.2
jupyter-lsp|2.2.4|2.2.4
jupyterlab|4.1.5|4.1.5
amazon-codewhisperer-jupyterlab-ext|2.0.1|2.0.1
jupyter-scheduler|2.5.1|2.5.1
amazon-sagemaker-jupyter-scheduler|3.0.7|3.0.7
${color{red}amazon-sagemaker-sql-magic}$|0.1.0|0.1.1
jupyterlab-lsp|5.0.3|5.0.3
${color{red}amazon_sagemaker_sql_editor}$|0.1.3|0.1.4
scipy|1.11.4|1.11.4
scikit-learn|1.4.1.post1|1.4.1.post1
pip|23.3.2|23.3.2
torchvision|0.15.2|0.15.2
autogluon|0.8.2|0.8.2
ipywidgets|8.1.2|8.1.2
notebook|7.1.2|7.1.2
aws-glue-sessions|1.0.4|1.0.4
conda|23.11.0|23.11.0
fastapi|0.103.2|0.103.2
${color{red}langchain}$|0.1.9|0.1.13
jupyter-ai|2.11.0|2.11.0
jupyter-dash|0.4.2|0.4.2
jupyter-server-proxy|4.1.2|4.1.2
jupyterlab-git|0.50.0|0.50.0
keras|2.12.0|2.12.0
matplotlib|3.8.3|3.8.3
nodejs|18.19.0|18.19.0
py-xgboost-cpu|1.7.6|1.7.6
thrift_sasl|0.4.3|0.4.3
pyhive|0.7.0|0.7.0
python-gssapi|1.8.3|1.8.3
python-lsp-server|1.10.1|1.10.1
sagemaker-code-editor|1.0.0|1.0.0
sagemaker-headless-execution-driver|0.0.12|0.0.12
sagemaker-jupyterlab-emr-extension|0.1.9|0.1.9
sagemaker-jupyterlab-extension|0.2.0|0.2.0
sagemaker-kernel-wrapper|0.0.2|0.0.2
sagemaker-python-sdk|2.212.0|2.212.0
sagemaker-studio-analytics-extension|0.0.21|0.0.21
sasl|0.3.1|0.3.1
supervisor|4.2.5|4.2.5
tensorflow|2.12.1|2.12.1
uvicorn|0.28.1|0.28.1

Comment (claytonparnell) [reply src/package_staleness.py:66]: Yeah we should keep the backslash there. Can find a different way to stop that warning, removing it causes the formatting to be all messed up for stale ones

**Review by claytonparnell — APPROVED** [review]

**Review by dylanraws — APPROVED** [review]
